src/zos_files/zowe/zos_files_for_zowe_sdk/datasets.py

```diff
@@ -11,13 +11,216 @@
+                raise ValueError
+        self.__primary = primary
+
+    @property
+    def secondary(self) -> Optional[int]:
+        return self.__secondary
+
+    @secondary.setter
+    def secondary(self, secondary: Optional[int]):
+        if self.primary is not None:
+            secondary = secondary if secondary is not None else int(self.primary / 10)
+            if secondary > 16777215:
+                raise ValueError
+            self.__secondary = secondary
+
+    @property
+    def dirblk(self) -> Optional[int]:
+        return self.__dirblk
+
+    @dirblk.setter
+    def dirblk(self, dirblk: Optional[int]):
+        self.__dirblk = dirblk
+
+    @property
+    def avgblk(self) -> Optional[int]:
+        return self.__avgblk
+
+    @avgblk.setter
+    def avgblk(self, avgblk: Optional[int]):
+        self.__avgblk = avgblk
+
+    @property
+    def recfm(self) -> Optional[str]:
+        return self.__recfm
+
+    @recfm.setter
+    def recfm(self, recfm: Optional[str]):
+        if recfm is None:
+            if self.like is not None:
+                self.__recfm = None
+            else:
+                self.__recfm = "F"
+        elif recfm not in ("F", "FB", "V", "VB", "U", "FBA", "FBM", "VBA", "VBM"):
+            raise KeyError
+        else:
+            self.__recfm = recfm
+
+    @property
+    def blksize(self) -> Optional[int]:
+        return self.__blksize
+
+    @blksize.setter
+    def blksize(self, blksize: Optional[int]):
+        if blksize is None:
+            if self.like is not None:
+                self.__blksize = None
+            else:
+                if self.lrecl is not None:
+                    self.__blksize = self.lrecl
+        else:
+            self.__blksize = blksize
+
+    @property
+    def lrecl(self) -> Optional[int]:
+        return self.__lrecl
+
+    @lrecl.setter
+    def lrecl(self, lrecl: Optional[int]):
+        self.__lrecl = lrecl
+
+    @property
+    def storclass(self) -> Optional[str]:
+        return self.__storclass
+
+    @storclass.setter
+    def storclass(self, storclass: Optional[str]):
+        self.__storclass = storclass
+
+    @property
+    def mgntclass(self) -> Optional[str]:
+        return self.__mgntclass
+
+    @mgntclass.setter
+    def mgntclass(self, mgntclass: Optional[str]):
+        self.__mgntclass = mgntclass
+
+    @property
+    def dataclass(self) -> Optional[str]:
+        return self.__dataclass
+
+    @dataclass.setter
+    def dataclass(self, dataclass: Optional[str]):
+        self.__dataclass = dataclass
+
+    @property
+    def dsntype(self) -> Optional[str]:
+        return self.__dsntype
+
+    @dsntype.setter
+    def dsntype(self, dsntype: Optional[str]):
+        self.__dsntype = dsntype
+
+    @property
+    def like(self) -> Optional[str]:
+        return self.__like
+
+    def to_dict(self) -> dict:
+        return {key.replace("_DatasetOption__", ""): value for key, value in self.__dict__.items() if value is not None}
```

Comment:
Curious why we need to replace `_DatasetOption__` here?

Reply:
Without raplacing "_DatasetOption__", the response json would be like: {"_DatasetOption__like": "None"}, potentially breaking some tests or documented format.